### PR TITLE
Added SurveyExecutionState Model

### PR DESCRIPTION
**Pull request by chrisjluc**
Survey Execution State model references `respondent` and the current `question` that they're on.

Issue: https://github.com/uwblueprint/somo/issues/10

### CODE REVIEW

**Comment by dinahshi** [comment]
Why should this be a separate table instead of a column on the respondent table? Are there plans for what we might put in this model in the future? `SurveyExecutionState` seems a little misleading if the only reference is to current question.

**Comment by chrisjluc** [comment]
I think they should be separate because the idea of `survey state` and `respondent` are conceptually different. 

The respondent should contain all information about the respondent, similar to account information, such as `location`, `age`, `company`.

The survey execution state contains everything to do with running the survey, like `start_time`, `time_since_last_response`, etc. It doesn't make sense to have these fields in respondents because it has no meaning, unless it's associated with a survey state.

You're correct that for now all we need is question reference - this allows us to have flexibility and modularity in the future and be clear where new fields should go instead of putting everything in respondent.

File: app/models/survey_execution_state.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# Table name: survey_execution_states
+#
+#  id                                   :integer          not null, primary key
+#  question_id                          :integer
+#  respondent_id                        :integer
+#  created_at                           :datetime         not null
+#  updated_at                           :datetime         not null
+#
+
+class SurveyExecutionState < ActiveRecord::Base
```
Comment (dinahshi) [inline app/models/survey_execution_state.rb:13]: Would prefer to name this `SurveyState`, execution seems redundant.

File: app/models/survey_execution_state.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  belongs_to :respondent
+  belongs_to :question
```
Comment (dinahshi) [inline app/models/survey_execution_state.rb:15]: Given that the model name has the word survey in it, might make sense to expose the associated survey. (More of a nice-to-have)

**Review by dinahshi — APPROVED** [review]
Reasoning makes sense to me. Two minor comments that I'll leave to your discretion.

File: app/models/survey_execution_state.rb (hunk 1 above)
Comment (chrisjluc) [reply app/models/survey_execution_state.rb:13]: It maybe ambiguous with drafting a survey or state of a survey itself. When I hear `SurveyState`, I don't think of the survey execution like we do.

I'm open to name suggestions though and discussion @katrinacrisostomo

**Pull request merged**